fpga/lib/psmake/petalinux.mk

```diff
@@ -206,6 +206,11 @@ $(PRJ_HDF): $(HDF) $(subst gethdf,FORCE,$(findstring gethdf,$(MAKECMDGOALS)))
 	mkdir -p $(dir $(TMPHDF))
 	ln -sf $(realpath $(HDF)) $(TMPHDF)
 	petalinux-config $(GEN_ARGS) --get-hw-description $(dir $(TMPHDF)) $(SILENTCONFIG)
+	# Linuxptp patch fix for issue here: 
+	# https://support.xilinx.com/s/question/0D52E00006ihQMPSA2/issues-adding-linuxptp-package-in-petalinux-20211?language=en_US
+	rm -f components/yocto/layers/meta-openembedded/meta-oe/recipes-connectivity/linuxptp/linuxptp_3.1.bb
+	cp ../../patches/linuxptp_3.1.1.bb components/yocto/layers/meta-openembedded/meta-oe/recipes-connectivity/linuxptp/
+
```

Comment:
Everything in fpga/lib has its upstream in some other repo.  So, this change should either be applied upstream to https://github.com/missinglinkelectronics/psmake, or it should be applied in a different way.  And that relative path is a problem, since it looks like it only applies when building for the ZCU106.  Also, it's not clear if this patch is necessary - I built the design just fine without this patch.

Reply:
Another aspect is, that changes to `.bb` files should be done using a custom bitbake layer. The typical one which always exists (or is created automatically on creating a PetaLinux project) is `project-spec/meta-user/`. In corundum we have another, corundum-specific layer, which is `meta-corundum/` at the top-level. So, if there is a need to modify a recipe (`.bb` file) for all PetaLinux projects in the corundum repo, I'd say that a recipe modification (a `.bbappend` or a `.bb` for a custom version of a software package) should go there.

Modifying any files in the otherwise completely generated area of `components/` of a PetaLinux project does not sound like a good idea, since you never know if and when the toolchain is overwriting that again.

I also think it is not a good idea to add this rather specific code to toolchain code, `psmake.mk` is this case, since it seems to be something very specific to a certain PetaLinux project and is definitively not needed in other PetaLinux projects.